projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -333,74 +312,22 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
    * Private method to update the visible previews navigating to the left (previous).
    */
   private previous(): void {
-    if (!this.images) {
+    if (!this.images || !this.previewConfig) {
       throw new Error('Internal library error - images must be defined');
     }
-    // check if prevImage should be blocked
-    if (this.isPreventSliding(0)) {
-      return;
-    }
-
-    if (this.start === 0) {
-      return;
+    if(this.start <= 0) {
+      // check if prevImage should be blocked
+      const preventSliding = !!this.slideConfig && this.slideConfig.infinite === false;
+      if(preventSliding) {
+        return;
+      }
+      this.end = this.images.length;
+    } else {
+      this.end--;
     }
-
-    this.start = Math.max(this.start - 1, 0);
-    this.end = Math.min(this.end - 1, this.images.length);
+    this.start = this.end - Math.min((this.previewConfig.number as number), this.images.length);
```

Comment:
Also, start and end are calculated in a co-dependant way, like in setIndexesPreviews().